### ISSUE TITLE
A de-correlation model for feature exclusion

### ISSUE THREAD

**Issue by FredrikKarlssonSpeech** [alan-turing-institute/MLJ.jl#1047]
**Is your feature request related to a problem? Please describe.**
Not all model frameworks handle highly correlated predictors well.

**Describe the solution you'd like**
A de-correlation model similar to the standardizer would be great to have for these cases, that would remove superfluous predictors before model training.

**Describe alternatives you've considered**
There seems to be none available in MLJ, except manually examining the correlations of predictors and removing identified not needed predictors using FreatureSelector.

**Additional context**

**Comment by ablaom** [comment]
Correct. MLJ does not directly provide such a transformer. Ideally, this would be more than a `Static` transformer: You want to learn the redundant features by just looking at training data, and then apply the learned filter to arbitrary new data.  

As this is  feature selection, worth mentioning #70 and #426.
